### DISCUSSION TITLE
Mandatory Cloud Connection

### DISCUSSION THREAD

**Discussion by wavemop** [petretiandrea/home-assistant-tapo-p100#675] [Q&A]
@petretiandrea 
hey, you wrote 

> "Although the integration works using LAN, the tapo device needs internet access to synchronize with tapo cloud, especially for credentials, a missing internet access could lead into "Invalid authentication error".

Is there any chance to disentangle this in the future or is this behaviour hardcoded in these smart home devices like tapo, hence there's no chance to ever be "unclouded" ?

**Comment by petretiandrea** [comment]
Honestly, I don't know the logics of tapo devices, so you can try to disable internet connection and if not work re-enable it
